### PR TITLE
Log subscription payload errors

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
## Scope of the PR

Added Sentry & error logging for a case when we have an error on Saleor GraphQL subscription. It should help us in spotting issues.

## Checklist

- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/avatax/src/lib/error-utils.ts [hunk 1]
```diff
@@ -1,13 +1,53 @@
...
+export class SubscriptionPayloadErrorChecker {
+  static SubscriptionPayloadError = BaseError.subclass("SubscriptionPayloadError");
+
+  constructor(
+    private injectedLogger: Pick<typeof logger, "error">,
+    private injectedErrorCapture: (expection: any) => void,
+  ) {}
+
+  checkPayload(payload: CalculateTaxesPayload | OrderCancelledPayload | OrderConfirmedPayload) {
+    // @ts-expect-error errors field is not yet typed on subscription payloads
```
Comment (witoszekdev) [inline apps/avatax/src/lib/error-utils.ts:26]: question: is this a missing feature in Saleor types or in codegen?

**Review by witoszekdev — DISMISSED** [review]

File: apps/avatax/src/lib/error-utils.ts (hunk 1 above)
Comment (krzysztofzuraw) [reply apps/avatax/src/lib/error-utils.ts:26]: Saleor types 😞

File: apps/avatax/src/lib/error-utils.ts [hunk 2]
```diff
@@ -1,13 +1,53 @@
...
 
-export const errorUtils = {
-  resolveTrpcClientError,
-};
+export class SubscriptionPayloadErrorChecker {
+  static SubscriptionPayloadError = BaseError.subclass("SubscriptionPayloadError");
+
+  constructor(
+    private injectedLogger: Pick<typeof logger, "error">,
+    private injectedErrorCapture: (expection: any) => void,
```
Comment (lkostrowski) [inline apps/avatax/src/lib/error-utils.ts:22]: - "exception" typo
- error is typed, so probably you should replace "any" with instance of this error

**Review by lkostrowski — APPROVED** [review]
non blocking

**Pull request merged**